src/domain/shared/errors.ts

```diff
@@ -9,8 +9,14 @@ export const RankedErrorLevel = [ErrorLevel.Info, ErrorLevel.Warn, ErrorLevel.Cr
 export class DomainError extends Error {
   name: string
   level?: ErrorLevel
-  constructor(message?: string) {
-    super(message)
+  constructor(message?: string | unknown | Error) {
+    if (typeof message === "string" || typeof message === "undefined") {
+      super(message)
+    } else if (message instanceof Error) {
+      super(message.message)
+    } else {
+      super()
```

Comment:
we shouldn't eat the error entirely. Pass some kind of stringified error or at least a message saying `the provided error had a type we couldn't handle`